test/e2e-upgrade-tests.sh

```diff
@@ -56,6 +56,7 @@ function install_head() {
 }
 
 function knative_setup() {
+  build_knative_from_source
```

Comment:
I suggest adding a comment here, like:

# Build Knative to generate Istio manifests from HEAD for install_latest_release
# We do it here because it's a one-time setup